### PR TITLE
Add ConsoleExternalLogLink integration test

### PR DESCRIPTION
**Pull request by rhamilto**
Resolves https://jira.coreos.com/browse/CONSOLE-1640

### CODE REVIEW

File: frontend/public/components/utils/link.tsx [hunk 1]
```diff
@@ -76,4 +76,5 @@ type ExternalLinkProps = {
   href: string;
   text: string;
   additionalClassName?: string;
+  dataTestId?: string;
```
Comment (spadgett) [inline frontend/public/components/utils/link.tsx]:
```suggestion
  dataTestID?: string;
```

File: frontend/integration-tests/tests/crd-extensions.scenario.ts [hunk 2]
```diff
@@ -199,4 +199,91 @@ describe('CRD extensions', () => {
...
+  describe('ConsoleExternalLogLink CRD', () => {
+    const crd = 'ConsoleExternalLogLink';
+    const name = `${testName}-cell`;
+    const podName = `${testName}-pod`;
+    const cell = $(`[data-test-id=${name}]`);
+    const text = `${name} Logs`;
+    const namespaceFilter = '^openshift-';
+
+    it(`displays YAML editor for creating a new ${crd} instance`, async() => {
+      await browser.get(`${appHost}/k8s/cluster/customresourcedefinitions?name=${crd}`);
```
Comment (spadgett) [inline frontend/integration-tests/tests/crd-extensions.scenario.ts]: I feel like we should have a utility function for creating things from YAML, so it's not repeated in so many places. But not for this PR.

**Comment by rhamilto** [comment]
Comment addressed, @spadgett.

File: frontend/public/components/utils/link.tsx [hunk 3]
```diff
@@ -66,7 +66,7 @@ export const getURLSearchParams = () => {
   return all;
 };
 
-export const ExternalLink: React.FC<ExternalLinkProps> = ({href, text, additionalClassName=''}) => <a className={classNames('co-external-link', additionalClassName)} href={href} target="_blank" rel="noopener noreferrer">{text}</a>;
+export const ExternalLink: React.FC<ExternalLinkProps> = ({href, text, additionalClassName='', dataTestID=''}) => <a className={classNames('co-external-link', additionalClassName)} href={href} target="_blank" rel="noopener noreferrer" data-test-id={dataTestID}>{text}</a>;
```
Comment (spadgett) [inline frontend/public/components/utils/link.tsx]: Does this mean there will be a `data-test-id=""` attribute on every link? Should we leave this `undefined` if not set? (I'm not sure React's behavior offhand here.

**Comment by rhamilto** [comment]
Round 2 comments addressed, @spadgett.

**Comment by rhamilto** [comment]
/hold

The new tests are flaking in CI.  Not sure why...

File: frontend/integration-tests/tests/crd-extensions.scenario.ts [hunk 4]
```diff
@@ -199,4 +199,91 @@ describe('CRD extensions', () => {
...
+      expect(crudView.resourceTitle.getText()).toEqual(name);
+    });
+
+    it(`creates a new test pod to display the ${crd} instance`, async() => {
+      await browser.get(`${appHost}/k8s/ns/${testName}/pods`);
+      await crudView.isLoaded();
+      await crudView.createYAMLButton.click();
+      await yamlView.isLoaded();
+      const content = await yamlView.getEditorContent();
+      const newContent = _.defaultsDeep({}, {metadata: {name: podName}}, safeLoad(content));
```
Comment (spadgett) [inline frontend/integration-tests/tests/crd-extensions.scenario.ts]: Does this fix the tests? It looks like there's another test not deleting a workload that's adopting and killing your pod.

```suggestion
      const newContent = _.defaultsDeep({}, {metadata: {name: podName, labels: {app: example}}}, safeLoad(content));
```

**Pull request merged**